### ISSUE TITLE
SC2034 - false positive with `$()` and `declare -F`

### ISSUE THREAD

**Issue by XVilka** [koalaman/shellcheck#2061]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

# Remove all shell functions
function killfuncs() {
	BASHFCNS=$(declare -F | cut -f3 -d" ")
	unset "$(BASHFCNS)"
}
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 5:
        BASHFCNS=$(declare -F | cut -f3 -d" ")
        ^-- SC2034: BASHFCNS appears unused. Verify use (or export if used externally).

$ 
```

#### Here's what I wanted or expected to see:

No warning

**Comment by koalaman** [comment]
ShellCheck is right, this variable is unused because you are calling it as a function instead of expanding it as a variable. What you wanted was `unset $BASHFCNS`
